pkg/postage/listener/listener.go

```diff
@@ -198,6 +208,11 @@ func (l *listener) sync(from uint64, updater postage.EventUpdater) error {
 			}
 		}
 
+		err = updater.UpdateBlockNumber(to)
```

Comment:
I think that moving the block height to the actual individual event signatures on `EventUpdater` and using the `Log.BlockNumber` (and thus updating the chain state on every write in batch service) would be more consistent. Then we can avoid this method altogether. WDYT?

Reply:
the advantage of this approach is that the block number is only updated after the entire block has been processed. whereas if its on individual events it's already updated after the first event. if the bee node stops/crashes mid-processing a block it will reprocess some events but that should be fine as all of them should be idempotent (at least better than skipping anything).